File: app/src/main/java/space/pal/sig/old/model/database/ApodDao.java
package space.pal.sig.old.model.database;

import androidx.lifecycle.LiveData;
import androidx.paging.DataSource;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import java.util.List;

import io.reactivex.Completable;
import space.pal.sig.old.model.Apod;

/**
 * SpaceNow
 * Created by Catalin on 7/24/2020
 **/
@Dao
public interface ApodDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    Completable insert(Apod apod);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    Completable insert(Apod... apods);

    @Query("delete from apod")
    Completable delete();

    @Query("select * from apod where type = 'IMAGE' order by date desc")
    DataSource.Factory<Integer, Apod> findAllImages();

    @Query("select * from apod order by date desc limit :limit")
    LiveData<List<Apod>> findAllWithLimit(int limit);

    @Query("select * from apod where id = :id")
    LiveData<Apod> findById(String id);

    @Query("select * from apod where id = :id")
    Apod findByIdSync(String id);

}

File: app/src/main/java/space/pal/sig/database/dao/ApodDao.kt
package space.pal.sig.database.dao

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import space.pal.sig.model.entity.AstronomyPictureOfTheDay

/**
 * SpaceNow
 * Created by Catalin on 12/20/2020
 **/
@Dao
interface ApodDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun save(astronomyPictureOfTheDay: AstronomyPictureOfTheDay)

    @Query("SELECT * FROM astronomy_pictures_of_the_day WHERE date = :date")
    fun findByDate(date: String): LiveData<AstronomyPictureOfTheDay>

    @Query("SELECT * FROM astronomy_pictures_of_the_day ORDER BY date DESC LIMIT 1")
    fun findLatest(): LiveData<AstronomyPictureOfTheDay>

}